sdk/core/core-rest-pipeline/test/auxiliaryAuthenticationHeaderPolicy.spec.ts

```diff
@@ -42,20 +39,26 @@ describe("AuxiliaryAuthenticationHeaderPolicy", function () {
       request,
       status: 200,
     };
-    const next = sinon.stub<Parameters<SendRequest>, ReturnType<SendRequest>>();
-    next.resolves(successResponse);
+    const next = vi.fn<Parameters<SendRequest>, ReturnType<SendRequest>>();
+    next.mockResolvedValue(successResponse);
 
     const mockAuxiliaryAuthenticationHeaderPolicy = createAuxiliaryAuthenticationHeaderPolicy(
       tokenScopes,
       [mockCredential]
     );
     await mockAuxiliaryAuthenticationHeaderPolicy.sendRequest(request, next);
 
-    assert(
-      fakeGetToken.calledWith(tokenScopes, {
-        abortSignal: undefined,
-        tracingOptions: undefined,
-      }),
+    assert.deepStrictEqual(
+      fakeGetToken.mock.calls,
```

Comment:
nit: this is a little hard to parse, is there any equivalent to `assert(fakeGetToken).toHaveBeenCalledWith(tokenScopes, { ....etc })` ?

I'm not familiar with vitest... yet...

Reply:
vitest unfortunately leans into jest patterns, which lack most of the convenience methods that sinon provides. So all calls are dumped into method.calls as a two-dimensional array (an array of calls that is itself an array of arguments). It's not my favorite and I definitely think we could use some helpers.